src/sentry/static/sentry/app/views/eventsV2/savedQuery/index.tsx

```diff
@@ -319,13 +320,42 @@ class SavedQueryButtonGroup extends React.PureComponent<Props, State> {
 
   render() {
     const {organization} = this.props;
+
+    const renderDisabled = p => (
+      <Hovercard
+        body={
+          <FeatureDisabled
+            features={p.features}
+            hideHelpToggle
+            message={t('Discover queries are disabled')}
+            featureName={t('Discover queries')}
+          />
+        }
+      >
+        {p.children(p)}
+      </Hovercard>
+    );
+
+    const renderQueryButton = (renderFunc: (disabled: boolean) => JSX.Element | null) => {
```

Comment:
```suggestion
    const renderQueryButton = (renderFunc: (disabled: boolean) => React.ReactNode) => {
```

The React.ReactNode type includes JSX elements & null.